core/src/main/scala/scala/pickling/internal/DefaultPicklerRegistry.scala

```diff
@@ -52,7 +52,15 @@ final class DefaultPicklerRegistry(generator: RuntimePicklerGenerator) extends P
   override def registerUnpickler[T](key: String, p: Unpickler[T]): Unit =
     unpicklerMap.put(key, p)
 
-  /** Checks the existince of an unpickler. */
+  /** Checks the existence of a pickler ignoring the registered generators. */
+  override def lookupExistingPickler(key: String): Option[Pickler[_]] =
```

Comment:
Note: Generators are for type constructors and SHOULD be safe to call.  They do not involve runtime reflection.

E.g.  `List[_]` is impossible to register a pickler for, but you *can* register a pickler that handles it *if* it can lookup a pickler for element types.

Note:  I may be adding statically generated "generator" functions for types like `List[_]` where you leave some of the type constructor args "empty" and we get them passed in at runtime and do a runtime lookup (but not runtime reflection).

In any case, you shoudl be using lookupPickler/lookupUnpickler.